web-common/src/features/connectors/connectors-utils.ts

```diff
@@ -97,3 +99,19 @@ export function makeTablePreviewHref(
       return null;
   }
 }
+
+/**
+ * Determines the correct icon key for a connector based on its configuration.
+ * Special case: MotherDuck connectors use "motherduck" icon even though they have driver: duckdb
+ */
+export function getConnectorIconKeyForMotherDuck(
+  connector: V1AnalyzedConnector,
+): string {
+  // Special case: MotherDuck connectors use md: path prefix
```

Comment:
I like this kind of handling for special cases– the complexity is pushed down & made very explicit